alpaca/broker/client.py

```diff
@@ -216,7 +262,156 @@ def upload_cip_data_for_account_by_id(self, account_id: Union[UUID, str]):
 
     # ############################## ACCOUNT ACTIVITIES ################################# #
 
-    def get_activities_for_account(
-        self, account_id: Union[UUID, str], activity_filter: GetAccountActivitiesRequest
-    ) -> List[BaseActivity]:
-        pass
+    def get_account_activities(
+        self,
+        activity_filter: GetAccountActivitiesRequest,
+        max_items_limit: Optional[int] = None,
+        handle_pagination: Optional[PaginationType] = None,
+    ) -> Union[List[BaseActivity], Iterator[List[BaseActivity]]]:
+        """
+        Gets a list of Account activities, with various filtering options. Please see the documentation for
+        GetAccountActivitiesRequest for more information as to what filters are available.
+
+        The return type of this function is List[BaseActivity] however the list will contain concrete instances of one
+        of the child classes of BaseActivity, either TradeActivity or NonTradeActivity. It can be a mixed list depending
+        on what filtering criteria you pass through `activity_filter`
+
+
+        Args:
+            activity_filter (GetAccountActivitiesRequest): The various filtering fields you can specify to restrict
+              results
+            max_items_limit (Optional[int]): A maximum number of items to return over all for when handle_pagination is
+              of type `PaginationType.FULL`. Ignored otherwise.
+            handle_pagination (Optional[PaginationType]): What kind of pagination you want. If None then defaults to
+              `PaginationType.FULL`
+
+        Returns:
+            Union[List[BaseActivity], Iterator[List[BaseActivity]]]: Either a list or an Iterator of lists of
+              BaseActivity child classes
+        """
+
+        if handle_pagination is None:
+            handle_pagination = PaginationType.FULL
+
+        if handle_pagination != PaginationType.FULL and max_items_limit is not None:
+            raise ValueError(
+                "max_items_limit can only be specified for PaginationType.FULL"
+            )
+
+        # otherwise, user wants pagination so we grab an interator
+        iterator = self._get_account_activities_iterator(
+            activity_filter=activity_filter,
+            max_items_limit=max_items_limit,
+            mapping=lambda x: [self._parse_activity(x) for x in x],
```

Comment:
For improved readability, maybe we can use a different variable here for the iterating var.